sstables/index_reader.hh

```diff
@@ -781,6 +794,13 @@ public:
         }
     }
 
+    static use_caching caching_mode(const query::partition_slice& ps) {
+        if (ps.options.contains(query::partition_slice::option::bypass_cache)) {
+            return use_caching::none;
+        }
+        return (global_cache_index_pages) ? use_caching::global : use_caching::local;
```

Comment:
why the parenthesis around `global_cache_index_pages`, we have `operator const T&` for `updateable_value<T>`. quoting the value with parenthesis does not help in this regard, IMHO.

Reply:
Why wouldn't bypass_cache use local caching? It's very helpful to cache the last layers of the promoted index.

Reply:
streaming_reader doesn't bother me. There ratio of data streamed / readers created is very high, so the effects on the cache are amortized.

Reply:
I think you referred to seeking which happens when moving to next partition. And therefore index must move the cursor to first position of next partition, but that's either current or next index page. If current, it's already referenced by index reader and therefore is not evicted. If next, it's not loaded yet.

Reply:
> Yes the reader can be cached indefinitely and pin those index pages due to https://github.com/scylladb/scylladb/issues/1865 [1][2].

Well not indefinitely, the entries have a TTL. But it can be a long time, yes.